ophyd/sim.py

```diff
@@ -524,15 +527,15 @@ def position(self):
 
 class SynAxisEmptyHints(SynAxis):
     @property
-    def hints(self):
+    def hints(self) -> Hints:
         return {}
 
 
 class SynAxisNoHints(SynAxis):
     readback = Cpt(_ReadbackSignal, value=0.0, kind="omitted")
 
     @property
-    def hints(self):
+    def hints(self) -> Hints:
```

Comment:
If this raises an `AttributeError` then this should be type hinted `None`, or given no return type.

Reply:
I guess we could also use `NoReturn`. If the type checker is happy with that I'll make the change.